### ISSUE TITLE
cannot find macro llvm_asm in this scope

### ISSUE THREAD

**Issue by flyingyizi** [rust-embedded/critical-section#15]
.cargo/config.toml file content:

[build]
target = "avr-specs/avr-atmega328p.json"

[target.'cfg(target_arch = "avr")']
runner = "ravedude uno -cb 57600"

[unstable]
build-std = ["core,alloc"]

problem:
```
  Compiling critical-section v0.2.7
   Compiling hash32 v0.2.1
   Compiling nom v5.1.2
   Compiling fugit v0.3.6
error: cannot find macro `llvm_asm` in this scope
   --> C:\Users\atmel\.cargo\registry\src\mirrors.ustc.edu.cn-61ef6e0cd06fb9b8\critical-section-0.2.7\src\lib.rs:127:13
    |
127 |             llvm_asm!(
    |             ^^^^^^^^

error: cannot find macro `llvm_asm` in this scope
   --> C:\Users\atmel\.cargo\registry\src\mirrors.ustc.edu.cn-61ef6e0cd06fb9b8\critical-section-0.2.7\src\lib.rs:139:17
    |
139 |                 llvm_asm!("sei" :::: "volatile");
    |                 ^^^^^^^^

error[E0635]: unknown feature `llvm_asm`
 --> C:\Users\atmel\.cargo\registry\src\mirrors.ustc.edu.cn-61ef6e0cd06fb9b8\critical-section-0.2.7\src\lib.rs:3:42    
  |
3 | #![cfg_attr(target_arch = "avr", feature(llvm_asm))]
  |                                          ^^^^^^^^
```

The old llvm_asm! has been removed. Therefore, it doesn't compile on new rustc anymore.

**Comment by jannic** [comment]
This will be resolved with critical-section 1.0, as the implementations are removed from this crate: https://github.com/embassy-rs/critical-section/commit/a3cdfd8e27a8fd823492c1c04ab28dd1be82f1f3

**Comment by Dirbaio** [comment]
`critical-section` 1.0 is now out, which contains no actual implementations. Implementations should be in the arch support crates now, such as `avr-device`.
